src/EditorFeatures/Core.Cocoa/Snippets/CSharpSnippets/SnippetExpansionClient.cs

```diff
@@ -55,111 +43,20 @@ protected override ITrackingSpan InsertEmptyCommentAndGetEndPositionTrackingSpan
             return SubjectBuffer.CurrentSnapshot.CreateTrackingSpan(commentSpan, SpanTrackingMode.EdgeExclusive);
         }
 
-        public override IExpansionFunction GetExpansionFunction(XElement xmlFunctionNode, string fieldName)
+        public override IExpansionFunction? GetExpansionFunction(XElement xmlFunctionNode, string fieldName)
         {
             if (!TryGetSnippetFunctionInfo(xmlFunctionNode, out var snippetFunctionName, out var param))
             {
                 throw new ArgumentException();
             }
 
-            switch (snippetFunctionName)
-            {
-                case "SimpleTypeName":
-                    return new SnippetFunctionSimpleTypeName(this, TextView, SubjectBuffer, fieldName, param);
-                case "ClassName":
-                    return new SnippetFunctionClassName(this, TextView, SubjectBuffer, fieldName);
-                case "GenerateSwitchCases":
-                    return new SnippetFunctionGenerateSwitchCases(this, TextView, SubjectBuffer, fieldName, param);
-                default:
-                    return null;
-            }
-        }
-
-        internal override Document AddImports(
-            Document document, int position, XElement snippetNode,
-            bool placeSystemNamespaceFirst, bool allowInHiddenRegions, CancellationToken cancellationToken)
-        {
-            var importsNode = snippetNode.Element(XName.Get("Imports", snippetNode.Name.NamespaceName));
-            if (importsNode == null ||
-                !importsNode.HasElements)
-            {
-                return document;
-            }
-
-            var root = document.GetSyntaxRootSynchronously(cancellationToken);
-            var contextLocation = root.FindToken(position).Parent;
-
-            var newUsingDirectives = GetUsingDirectivesToAdd(contextLocation, snippetNode, importsNode, cancellationToken);
-            if (!newUsingDirectives.Any())
-            {
-                return document;
-            }
-
-            // In Venus/Razor, inserting imports statements into the subject buffer does not work.
-            // Instead, we add the imports through the contained language host.
-            if (TryAddImportsToContainedDocument(document, newUsingDirectives.Where(u => u.Alias == null).Select(u => u.Name.ToString())))
```

Comment:
If you're not looking commit-by-commit this might look odd, but `TryAddImportsToContainedDocument` used to always throw, therefore the rest of this method was unreachable, therefore this whole `AddImports` method never did anything.

I'm guessing there are no snippets that add usings in VS for Mac??

Reply:
@Therzok @DavidKarlas @KirillOsenkov Does that make sense? Is this an extension point for VS for Mac? (not that I can see how it worked if it is)

Reply:
I'm guessing it was copied from here but we had no impl due to MonoDevelopWorkspace being in MonoDevelop, not platform assemblies:
https://github.com/dotnet/roslyn/blob/cbf3129b85a84841941f6282e66560df8d62638f/src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs#L561

Reply:
@davidwengier yes that make sense, no snippet inserts usings so I think its fine to delete this.